### PR TITLE
add Spring Security

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/java/com/dev/cinema/model/dto/hall/CinemaHallRequestDto.java
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/model/dto/hall/CinemaHallRequestDto.java]: You should add message in bracket (message = "capacity shouldn't be null")

File: src/main/java/com/dev/cinema/validation/FieldsValueMatch.java
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/validation/FieldsValueMatch.java:23]: You don't need this block (20-23) - you can use it when you have several pairs to check (like password - repeatpassword, email - repeatEmail, etc)

**Review by fyarkiy — APPROVED** [review]

File: README.md
Comment (RomanDubovskyi) [inline README.md:2]: add description after you finish this project with single PR

File: src/main/java/com/dev/cinema/model/dto/hall/CinemaHallRequestDto.java
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/dto/hall/CinemaHallRequestDto.java]: can int be null?

File: src/main/java/com/dev/cinema/model/dto/movie/MovieRequestDto.java
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/dto/movie/MovieRequestDto.java]: ideally it's better to add some informative message

File: src/main/java/com/dev/cinema/model/dto/user/UserRegistrationDto.java
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/dto/user/UserRegistrationDto.java]: do we need list here?

File: src/main/java/com/dev/cinema/validation/EmailValidation.java
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/validation/EmailValidation.java]: do we need to put this annotation on method ?

File: src/main/java/com/dev/cinema/validation/EmailValidator.java
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/validation/EmailValidator.java]: it seems that we don't need this method

File: src/main/java/com/dev/cinema/validation/EmailValidator.java
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/validation/EmailValidator.java]:
```suggestion
    private static final String VALID_EMAIL_REGEX = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$";
```
regex is not informative

File: src/main/java/com/dev/cinema/validation/FieldsValueMatchValidator.java
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/validation/FieldsValueMatchValidator.java]: you can use Objects.equals() here instead

**Review by RomanDubovskyi — APPROVED** [review]
fix comments and you can merge without second review

File: src/main/java/com/dev/cinema/model/dto/hall/CinemaHallRequestDto.java
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/hall/CinemaHallRequestDto.java]: done

Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/hall/CinemaHallRequestDto.java]: done

File: src/main/java/com/dev/cinema/model/dto/movie/MovieRequestDto.java
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/movie/MovieRequestDto.java]: done

File: src/main/java/com/dev/cinema/model/dto/user/UserRegistrationDto.java
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/dto/user/UserRegistrationDto.java]: done

File: src/main/java/com/dev/cinema/validation/EmailValidation.java
Comment (vremere-a) [reply src/main/java/com/dev/cinema/validation/EmailValidation.java]: done

File: src/main/java/com/dev/cinema/validation/EmailValidator.java
Comment (vremere-a) [reply src/main/java/com/dev/cinema/validation/EmailValidator.java]: done

File: src/main/java/com/dev/cinema/validation/EmailValidator.java
Comment (vremere-a) [reply src/main/java/com/dev/cinema/validation/EmailValidator.java]: done

File: src/main/java/com/dev/cinema/validation/FieldsValueMatchValidator.java
Comment (vremere-a) [reply src/main/java/com/dev/cinema/validation/FieldsValueMatchValidator.java]: done

**Pull request merged**
